Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h

```diff
@@ -322,6 +322,8 @@ class UnifiedPDFPlugin final : public PDFPluginBase, public WebCore::GraphicsLay
     WebCore::IntPoint convertFromDocumentToPage(const WebCore::IntPoint&, PDFDocumentLayout::PageIndex) const;
     WebCore::IntPoint convertFromPageToDocument(const WebCore::IntPoint&, PDFDocumentLayout::PageIndex) const;
     WebCore::IntPoint convertFromPageToContents(const WebCore::IntPoint&, PDFDocumentLayout::PageIndex) const;
+    WebCore::FloatRect convertFromPageToContents(const WebCore::FloatRect&, PDFDocumentLayout::PageIndex) const;
```

Comment:
Weird that this one is floaty and the others aren't (though maybe they all should be?)

Humorously I also needed this one... I'll use yours if it lands first :)

Reply:
I considered Float-ing everything too. I have a patch where I do pageToDocument->documentToPlugin->pluginToRoot and that introduces too much rounding/truncation error.

Reply:
Interestingly all the Rect and Point objects have very similar interfaces and it would be v cool if we could write.

```
template <typename T>
concept IsRectOrPointLike = std::same_as<T, IntRect> || std::same_as<T, FloatRect> || std::same_as<T, IntPoint> || std::same_as<T, FloatPoint>

IsRectOrPointLike convertFromXtoY(const IsRectOrPointLike&);
```